fop/fom.c

```diff
@@ -634,8 +638,17 @@ M0_INTERNAL void m0_fom_queue(struct m0_fom *fom)
 	fom->fo_loc = dom->fd_localities[loc_idx];
 	fom->fo_loc_idx = loc_idx;
 	m0_fom_sm_init(fom);
+	addb2_introduce(fom);
+	m0_fom_locality_inc(fom);
 	fom->fo_cb.fc_ast.sa_cb = &queueit;
+	if (m0_reqh_service_state_get(fom->fo_service) == M0_RST_STOPPED ||
```

Comment:
queueit() and reqh shutdown  too is not synchronized.
This patch is just reducing the race window, looks like there is no way to avoid it without introduction of a RCU kind of lock here.
@nikitadanilov : Can you suggest the approach here ?

Reply:
This is changed in latest update.